mapboxgl/viz.py

```diff
@@ -345,3 +360,88 @@ def add_unique_template_variables(self, options):
             lineWidth=self.line_width,
         ))
 
+
+class AltChoroplethViz(MapViz):
+    """Create a choropleth viz"""
+
+    def __init__(self,
+                 data,
+                 vector_polygon_source=False,
+                 vector_url=None,
+                 vector_layer_name=None,
+                 vector_join_color_property=None,
```

Comment:
Could we merge this with the existing `color_property` class variable?

Reply:
I found this necessary for dealing with vector data that might have different property names than your joined geojson. For example, you have the US States vector source, which has a property called STATEFP, which is the "id" and our geojson object, with a property called "id". If there is a way to override vector property names I'd love to hear it!! Rather than rely on users to always have well-formatted geojson, I thought this would provide more flexibility.